### ISSUE TITLE
[no-unused-var] false positive with nested namespace and named function expression

### ISSUE THREAD

**Issue by unional** [typescript-eslint/typescript-eslint#1917]
**Repro**
```ts
export namespace foo.config {
  export type Options = {}
}
export function foo() { }
foo.config = function config(options: foo.config.Options) { return options }
```

**Expected Result**
no error

**Actual Result**
![image](https://user-images.githubusercontent.com/3254987/79645294-5372cf80-8163-11ea-9a6f-0a804f531e44.png)

**Additional Info**
The following works fine

```ts
export namespace foo.config {
  export type Options = {}
}
export function foo() { }
// anonymous function expression
foo.config = function (options: foo.config.Options) { return options }
```
```ts
export namespace foo.config {
  export type Options = {}
}
export function foo() { }
// function declaration
function config(options: foo.config.Options) { return options }
foo.config = config
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.0` |
| `@typescript-eslint/parser`        | `2.28.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
See #1856
